dlib/CMakeLists.txt

```diff
@@ -601,6 +606,15 @@ if (NOT TARGET dlib)
          endif()
       endif()
 
+      include(cmake_utils/find_ffmpeg.cmake)
+      if (DLIB_USE_FFMPEG)
+         if (FFMPEG_FOUND)
+            set (dlib_needed_libraries ${dlib_needed_libraries} ${FFMPEG_LIBRARIES})
+            set (dlib_needed_includes ${dlib_needed_includes} ${FFMPEG_INCLUDE_DIRS})
```

Comment:
@arrufat Isn't it ```${libavdevice_CFLAGS} ${libavfilter_CFLAGS} ${libavformat_CFLAGS} ${libavcodec_CFLAGS} ${libswresample_CFLAGS} ${libswscale_CFLAGS} ${libavutil_CFLAGS}```

Reply:
Likewise, what I suggested is wrong. But i think we need to use ```FFMPEG_CFLAGS```, not ```FFMPEG_INCLUDE_DIRS```

Reply:
I will have a look, thank you :)

Reply:
But your cmake file is way simpler than opencv's which is probably using older cmake functions. I think once the LDFLAGS and CFLAGS are sorted, it will work fine. The only other thing to do, which would be nice, is to have ```DLIB_FMFPEG_DIR``` which gets added to ```PKG_CONFIG_PATH``` somehow.

Reply:
CMAKE_PREFIX_PATH is usually the right thing to use for this kind of thing. The other thing you often do is discover the right path by using find_file, find_library, find_path or whatever to discover the right path.

Reply:
Okidok. I mean whatever works. But simply using ```-L<path to ffmpeg installation>/lib -lavdevice -lavformat -lavcodec ...``` will be insufficient in most cases. So I think using ```FFMPEG_LDFLAGS``` and ```FFMPEG_CFLAGS``` will be necessary instead of ```FFMPEG_LIBRARIES``` and ```FFMPEG_INCLUDE_DIRS``` if i understand cmake docs correctly.

Reply:
I'm not saying don't use pkg-config mind you.  Just that `DLIB_FMFPEG_DIR` shouldn't be needed.  `CMAKE_PREFIX_PATH` ought to cover it and then beyond that there can be some default search rules using `find_path()` say that look in normal places to find the right path to use with `find_package(PkgConfig)`.

Reply:
ah ok I see